### DISCUSSION TITLE
Cannot get STA-AP to work with piZero 2

### DISCUSSION THREAD

**Discussion by mcdullmedia** [RaspAP/raspap-webgui#1594] [General]
I have tried everything mentioned in the tutorial.  No reboot before enabling STA+AP mode.
The best I could get is zero2connected to my other AP as client, and AP mode visible.  
When my other device connecting it, no ip is assigned.  Even I manually entered IP, no internet on the client.
DHCP server is bind to uap0 but didnt work.

And rebooting the zero2 will randomly connect using wlan0 or uap0 (I have meshcentral installed so it tells which interface connecting to the host network)

 Tried to search for similar issues, solution not working.  Thank you.

**Comment by billz** [comment]
Per the AP-STA doc page:
> This mode is _completely unsupported_ and should be used for educational purposes only. If you need a reliable solution with an access point (AP) and wireless client (STA) on the same device, buy a second Wi-Fi adapter and follow [this FAQ](https://docs.raspap.com/faq/#interfaces) instead.

**Comment by evilsibling** [comment]
hey @mcdullmedia go to Hostpot and then Logging, what error(s) do you see.
On my Pi Zero2 W I get the below errors when I try to enable STA-AP which looks to me like either we need to specify a different driver to use for the uap0 interface, or the the Pi Zero2 W can't actually support this mode.

```
uap0: interface state UNINITIALIZED->COUNTRY_UPDATE
uap0: INTERFACE-DISABLED 
uap0: INTERFACE-ENABLED 
Could not set channel for kernel driver
Interface initialization failed
uap0: interface state COUNTRY_UPDATE->DISABLED
uap0: AP-DISABLED 
uap0: interface state DISABLED->DISABLED
uap0: AP-DISABLED 
uap0: CTRL-EVENT-TERMINATING 
hostapd_free_hapd_data: Interface uap0 wasn't started
nl80211: deinit ifname=uap0 disabled_11b_rates=0
Could not read interface uap0 flags: No such device
nl80211: Driver does not support authentication/association or connect commands
nl80211: deinit ifname=uap0 disabled_11b_rates=0
Could not read interface uap0 flags: No such device
nl80211 driver initialization failed.
uap0: interface state UNINITIALIZED->DISABLED
uap0: AP-DISABLED 
uap0: CTRL-EVENT-TERMINATING 
```

**Reply by evilsibling** [reply]
Ok it looks like i missed this note ( https://github.com/RaspAP/raspap-webgui/issues/752#issuecomment-757448664 ) about needing to "stop hostapd, disable AP-STA, make your config changes on wlan0, re-enable AP-STA and start hostapd.". Ive got STA-AP working on my Pi Zero2 W now
